### DISCUSSION TITLE
Tuya IR Blaster IH-F8260

### DISCUSSION THREAD

**Discussion by malaki86** [Koenkk/zigbee2mqtt#19881] [Devices]
I just received a Tuya-based IR Blaster. Z2M detects and pairs with it, but it is an unrecognizable device. It detects Zigbee model TS1201, and is ZS06 based. 

What do I need to do to fix this? I'm running the HACS installation in HAOS.

![Screenshot_20231126_101329_Home Assistant](https://github.com/Koenkk/zigbee2mqtt/assets/50643710/7b90987c-2185-4c75-8c4d-1d3a55d5fbf1)

**Comment by wontdieone** [comment]
I managed to get it working by adding external converter with the code below.
Cheers.

```
const {onOff} = require('zigbee-herdsman-converters/lib/modernExtend');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const {} = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const zosung = require('zigbee-herdsman-converters/lib/zosung');
const e = exposes.presets;
const ea = exposes.access;
const fzZosung = zosung.fzZosung;
const tzZosung = zosung.tzZosung;
const ez = zosung.presetsZosung;

const definition = {
    fingerprint: [
        {
            modelID: 'TS1201',
            manufacturerName: '_TZ3290_gnl5a6a5xvql7c2a',
        },
    ],
    model: 'iH-F8260',
    vendor: 'TuYa',
    description: 'Universal smart IR remote control',
	fromZigbee: [
		fzZosung.zosung_send_ir_code_00, fzZosung.zosung_send_ir_code_01, fzZosung.zosung_send_ir_code_02,
		fzZosung.zosung_send_ir_code_03, fzZosung.zosung_send_ir_code_04, fzZosung.zosung_send_ir_code_05,
	],
	toZigbee: [tzZosung.zosung_ir_code_to_send, tzZosung.zosung_learn_ir_code],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: tuya.configureMagicPacket,
    exposes: [ez.learn_ir_code(), ez.learned_ir_code(), ez.ir_code_to_send()],
};

module.exports = definition;
```

**Reply by jakubsz123** [reply]
Could you write the steps of how you did it, please? It's hard to find a guide on how to add the external converter

**Reply by wontdieone** [reply]
The guide to follow would be here: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
For TuYa devices would be here: https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html

1. Copy the code above into a file named "iH-F8260.js"
2. Copy the file "iH-F8260.js" into your zigbee2mqtt folder in home assistant OS server
3. In Z2M in home assistant, goto Settings -> External Converters -> + sign -> add in "iH-F8260"
4. Restart Z2M

**Comment by kikher** [comment]
Thanks to your code I have managed to support it z2m. Would you be able to tell me if you have marked it so that they can officially support it in the next update?

Greetings and thank you!

**Reply by wontdieone** [reply]
I have not marked it yet. Feel free to do so.

**Reply by kikher** [reply]
It's already uploaded in the dev. I imagine that in the next update it will be official!

**Comment by boeserwolf** [comment]
Thank you so much @wontdieone, I don't know why this device isn't supported officially, but thanks to your external converter I got it up & running.

Cheers & thank you again! 🙂
